app/components/autocomplete/at_mention/index.js

```diff
@@ -4,21 +4,29 @@
 import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 
-import {getTheme} from 'app/selectors/preferences';
 import {autocompleteUsers} from 'mattermost-redux/actions/users';
-import {getDefaultChannel} from 'mattermost-redux/selectors/entities/channels';
-import {getProfilesInCurrentChannel, getProfilesNotInCurrentChannel} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentChannelId, getDefaultChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+
+import {
+    filterMembersInChannel,
+    filterMembersNotInChannel,
+    filterMembersInCurrentTeam,
+    getMatchTermForAtMention
+} from 'app/selectors/autocomplete';
+import {getTheme} from 'app/selectors/preferences';
 
 import AtMention from './at_mention';
 
 function mapStateToProps(state, ownProps) {
-    const {currentChannelId} = state.entities.channels;
+    const {cursorPosition, isSearch, rootId} = ownProps;
+    const currentChannelId = getCurrentChannelId(state);
 
     let postDraft;
```

Comment:
Minor thing, but we could have the PostTextbox and Search screen pass their value into the Autocomplete where it'll be passed onto AtMention and ChannelMention so that we don't need to bother figuring out the draft here

Reply:
Yeah we can do that with the performance optimizations, cause we could even change the post draft being set on every key and set it only when we end typing

Reply:
Cool, I'll make a ticket